nan.h

```diff
@@ -1169,24 +1242,80 @@ NAN_INLINE _NanWeakCallbackInfo<T, P>* NanMakeWeakPersistent(
     return _NAN_ERROR(v8::Exception::TypeError, errmsg);
   }
 
-  NAN_INLINE v8::Local<v8::Value> NanThrowTypeError(
-      const char* errmsg
-  ) {
+  NAN_INLINE v8::Local<v8::Value> NanTypeError(v8::Handle<v8::String> errmsg) {
+    return v8::Exception::TypeError(errmsg);
+  }
+
+  NAN_INLINE v8::Local<v8::Value> NanThrowTypeError(const char* errmsg) {
     _NAN_THROW_ERROR(v8::Exception::TypeError, errmsg);
   }
 
+  NAN_INLINE
+  v8::Local<v8::Value> NanThrowTypeError(v8::Handle<v8::String> errmsg) {
+    NanEscapableScope();
+    return NanEscapeScope(v8::Local<v8::Value>::New(
+        v8::ThrowException(v8::Exception::TypeError(errmsg))));
+  }
+
+  NAN_INLINE v8::Local<v8::Value> NanSyntaxError(const char* errmsg) {
+    return _NAN_ERROR(v8::Exception::SyntaxError, errmsg);
+  }
+
+  NAN_INLINE v8::Local<v8::Value> NanSyntaxError(v8::Handle<v8::String> errmsg) {
+    return v8::Exception::SyntaxError(errmsg);
+  }
+
+  NAN_INLINE v8::Local<v8::Value> NanThrowSyntaxError(const char* errmsg) {
+    _NAN_THROW_ERROR(v8::Exception::SyntaxError, errmsg);
+  }
+
+  NAN_INLINE
+  v8::Local<v8::Value> NanThrowSyntaxError(v8::Handle<v8::String> errmsg) {
+    NanEscapableScope();
+    return NanEscapeScope(v8::Local<v8::Value>::New(
+        v8::ThrowException(v8::Exception::SyntaxError(errmsg))));
+  }
+
+  NAN_INLINE v8::Local<v8::Value> NanReferenceError(const char* errmsg) {
+    return _NAN_ERROR(v8::Exception::ReferenceError, errmsg);
+  }
+
+  NAN_INLINE v8::Local<v8::Value> NanReferenceError(v8::Handle<v8::String> errmsg) {
+    return v8::Exception::ReferenceError(errmsg);
+  }
+
+  NAN_INLINE v8::Local<v8::Value> NanThrowReferenceError(const char* errmsg) {
+    _NAN_THROW_ERROR(v8::Exception::ReferenceError, errmsg);
+  }
+
+  NAN_INLINE
+  v8::Local<v8::Value> NanThrowReferenceError(v8::Handle<v8::String> errmsg) {
+    NanEscapableScope();
+    return NanEscapeScope(v8::Local<v8::Value>::New(
+        v8::ThrowException(v8::Exception::ReferenceError(errmsg))));
+  }
+
   NAN_INLINE v8::Local<v8::Value> NanRangeError(
       const char* errmsg
   ) {
     return _NAN_ERROR(v8::Exception::RangeError, errmsg);
   }
 
-  NAN_INLINE v8::Local<v8::Value> NanThrowRangeError(
-      const char* errmsg
-  ) {
+  NAN_INLINE v8::Local<v8::Value> NanRangeError(v8::Handle<v8::String> errmsg) {
+    return v8::Exception::RangeError(errmsg);
+  }
+
+  NAN_INLINE v8::Local<v8::Value> NanThrowRangeError(const char* errmsg) {
     _NAN_THROW_ERROR(v8::Exception::RangeError, errmsg);
   }
 
+  NAN_INLINE
+  v8::Local<v8::Value> NanThrowRangeError(v8::Handle<v8::String> errmsg) {
+    NanEscapableScope();
+    return NanEscapeScope(v8::Local<v8::Value>::New(
+        v8::ThrowException(v8::Exception::RangeError(errmsg))));
+  }
+
```

Comment:
Okay.  It's not wrong to use a HandleScope here, it's just that the overhead is pretty high compared to the meat of the function.  That probably doesn't matter so much here, though.